deep-prove/src/bin/cli/main.rs

```diff
@@ -19,32 +24,49 @@ mod lagrange {
 #[derive(Parser)]
 #[command(version, about)]
 struct Args {
-    /// The URL of the Gateway to the proving network to connect to
-    #[clap(short, long, env)]
-    gw_url: Url,
+    #[command(subcommand)]
+    executor: Executor,
+}
 
-    /// The client ETH private key.
-    #[clap(short, long, env)]
-    private_key: String,
+#[derive(Subcommand)]
+enum Executor {
+    Lpn {
+        /// The URL of the Gateway to the proving network to connect to
+        #[clap(short, long, env)]
+        gw_url: Url,
 
-    /// Max message size passed through gRPC (in MBytes)
-    #[arg(long, default_value = "100")]
-    max_message_size: usize,
+        /// The client ETH private key.
+        #[clap(short, long, env)]
+        private_key: String,
```

Comment:
maybe lets use a file instead? Not sure how this will be used, but if it is part of the env or CLI arg it is super easy to leak it.